halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1508,6 +1623,31 @@ impl<F: Field, C: Into<Constraint<F>>, Iter: IntoIterator<Item = C>> IntoIterato
     }
 }
 
+/// GateV2Backend
+#[derive(Clone, Debug)]
+pub struct GateV2Backend<F: Field> {
+    name: String,
+    constraint_names: Vec<String>,
+    polys: Vec<ExpressionMid<F>>,
+}
+
+impl<F: Field> GateV2Backend<F> {
+    /// Returns the gate name.
+    pub fn name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    /// Returns the name of the constraint at index `constraint_index`.
+    pub fn constraint_name(&self, constraint_index: usize) -> &str {
+        self.constraint_names[constraint_index].as_str()
+    }
+
+    /// Returns constraints of this gate
+    pub fn polynomials(&self) -> &[ExpressionMid<F>] {
```

Comment:
I'd say this might be a better fit? Otherwise, you'd not send a `Vec<ExpressionMid<F>>` rather a `Polynomial` structure.

```suggestion
    pub fn expressions(&self) -> &[ExpressionMid<F>] {
```

Also, the one I prefer is `identities` which probabli fits the best. As each item of the `Vec` returned in `create_gate` fn is called an `identity`.
```suggestion
    pub fn identities(&self) -> &[ExpressionMid<F>] {
```

Reply:
I didn't change anything here https://github.com/privacy-scaling-explorations/halo2/blob/73408a140737d8336490452193b21f5a7a94e7de/halo2_proofs/src/plonk/circuit.rs#L1535

I think for code that is not changed, it's better to keep the same name (even if different names could make more sense) to make it easier to map the code to halo2 upstream and other forks.